components/org.wso2.identity.event.websubhub.publisher/src/main/java/org/wso2/identity/event/websubhub/publisher/internal/ClientManager.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.identity.event.websubhub.publisher.internal;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.conn.ssl.DefaultHostnameVerifier;
+import org.apache.http.impl.nio.client.CloseableHttpAsyncClient;
+import org.apache.http.impl.nio.client.HttpAsyncClientBuilder;
+import org.apache.http.impl.nio.client.HttpAsyncClients;
+import org.apache.http.impl.nio.conn.PoolingNHttpClientConnectionManager;
+import org.apache.http.impl.nio.reactor.DefaultConnectingIOReactor;
+import org.apache.http.nio.reactor.ConnectingIOReactor;
+import org.apache.http.ssl.SSLContexts;
+import org.wso2.identity.event.websubhub.publisher.constant.WebSubHubAdapterConstants;
+import org.wso2.identity.event.websubhub.publisher.exception.WebSubAdapterException;
+import org.wso2.identity.event.websubhub.publisher.util.WebSubHubAdapterUtil;
+
+import java.io.IOException;
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+
+import javax.net.ssl.SSLContext;
+
+import static java.util.Objects.isNull;
+
+/**
+ * Class to retrieve the HTTP Clients.
+ */
+public class ClientManager {
+
+    private static final Log LOG = LogFactory.getLog(ClientManager.class);
+    private final CloseableHttpAsyncClient httpAsyncClient;
+
+    /**
+     * Creates a client manager.
+     *
+     * @throws WebSubAdapterException on errors while creating the http client.
+     */
+    public ClientManager() throws WebSubAdapterException {
+
+        LOG.info("Initializing ClientManager");
+        PoolingNHttpClientConnectionManager connectionManager;
+        try {
+            connectionManager = createPoolingConnectionManager();
+            LOG.info("Successfully created PoolingNHttpClientConnectionManager");
```

Comment:
IMO, we don't need this info log. Let's make it a debug

Reply:
Addressed with https://github.com/wso2-extensions/identity-event-publishers/pull/1/commits/f0ff52527687246da83cb9dbcb13ce640f0d6db3